ext/span.c

```diff
@@ -548,6 +636,16 @@ void ddtrace_close_span(ddtrace_span_data *span) {
         ddtrace_switch_span_stack(span->stack);
     }
 
+    if (span->std.ce == ddtrace_ce_root_span_data) {
+        ddtrace_root_span_data *root = ROOTSPANDATA(&span->std);
+        zval *inferred_span_zv = &root->property_inferred_span;
+        if (Z_TYPE_P(inferred_span_zv) == IS_OBJECT) {
```

Comment:
could you please add a condition that the inferred_span property is actually an instance of ddtrace_ce_inferred_span_data?